yarn-project/pxe/src/contract_function_simulator/execution_note_cache.ts

```diff
@@ -173,12 +173,14 @@ export class ExecutionNoteCache {
    * Return notes created up to current point in execution.
    * If a nullifier for a note in this list is emitted, the note will be deleted.
    * @param contractAddress - Contract address of the notes.
-   * @param owner - Owner of the notes.
+   * @param owner - Owner of the notes. If undefined, returns all notes regardless of owner.
    * @param storageSlot - Storage slot of the notes.
    **/
-  public getNotes(contractAddress: AztecAddress, owner: AztecAddress, storageSlot: Fr) {
+  public getNotes(contractAddress: AztecAddress, owner: AztecAddress | undefined, storageSlot: Fr) {
     const notes = this.noteMap.get(contractAddress.toBigInt()) ?? [];
-    return notes.filter(n => n.note.owner.equals(owner) && n.note.storageSlot.equals(storageSlot)).map(n => n.note);
+    return notes
+      .filter(n => (owner === undefined || n.note.owner.equals(owner)) && n.note.storageSlot.equals(storageSlot))
```

Comment:
```suggestion
      .filter(n => owner === undefined || n.note.owner.equals(owner))
      .filter(n => n.note.storageSlot.equals(storageSlot))
```

Nit, it's free to have more filtering steps (might even be faster), and I think this is much more readable.